src/shuffler-codereview.c

```diff
@@ -0,0 +1,1228 @@
+        // and check pool2. if it also exceeds the bounds of pool2, subtract the
+        // size of pool2 and check pool3. right now there's only three pools, so
+        // anything beyond that is an error.
+        if (partyIndicies[i] < tt->pool1Size) {
+            pkmn = &tt->pool1[partyIndicies[i]];
+        } else if (partyIndicies[i] < tt->pool1Size + tt->pool2Size) {
+            pkmn = &tt->pool2[partyIndicies[i] - tt->pool1Size];
+        } else if (partyIndicies[i] < tt->pool1Size + tt->pool2Size + tt->pool3Size) {
+            pkmn = &tt->pool3[partyIndicies[i] - tt->pool1Size - tt->pool2Size];
+        } else {
+            MYLOG("partyIndicies[%d] (%d) exceeds all three pools of sizes %d, %d, %d",
+                i, partyIndicies[i], tt->pool1Size, tt->pool2Size, tt->pool3Size);
+        }
+        // evolve the species if appropriate. 
+        partyMember->species = AdjustSpecies(pkmn->species, level, pkmn->evoStrat);
+
+        // we always explicitly set the item, even if it's ITEM_NONE, which is
+        // the default.
+        partyMember->heldItem = pkmn->heldItem;
+
+        // fill the pokemon's moves. our trainer format allows us to specify
+        // a partial moveset and fill in remaining spots with whatever the
+        // pokemon would natrually learn at their level, or to leave some move
+        // spots blank.
+        // if the first move is set to MOVE_NONE, we explicitly set all the 
+        // moves to MOVE_UNAVAILABLE so they'll be filled in with the moves the
+        // pokemon would natrually learn. if no moves are specified, the data
+        // at this location will be 0 by default, which is equivilant to
+        // MOVE_NONE. if the first move is anything else, we treat values of
+        // MOVE_UNAVAILABLE as moves to be filled in by natrually learned
+        // moves, and we treat MOVE_NONE as intentionally blank spots.
+        if (pkmn->moves[0] == MOVE_NONE) {
+            for (int j = 0; j < MAX_MON_MOVES; j++) {
+                partyMember->moves[j] = MOVE_UNAVAILABLE;
+            }
+        } else {
+            for (int j = 0; j < MAX_MON_MOVES; j++) {
+                partyMember->moves[j] = pkmn->moves[j];
+            }
+        }
+    }
+    trainer->party.ItemCustomMoves = trainerInfo->party;
+
+    trainerInfo->defeatText = tt->defeatText;
+    trainerInfo->introText = tt->introText;
+    trainerInfo->postbattleText = tt->postbattleText;
+    trainerInfo->name = tt->trainerName;
+    AdjustedTemplates[objNum].graphicsId = tt->graphicsId;
+    AdjustedTemplates[objNum].flagId = ShuffledFlagNumberByObjectEventId(objNum + 1);
+
+    // check to see if we have a partner object and see if that partner object
+    // has already been generated. if it has, we need to combine the two
+    // objects so the player gets the same battle when interacting with either
+    // one.
+    if (partnerObjNum != 0xFF && (AdjustedObjectsDirty & (1 << partnerObjNum))) {
```

Comment:
condition smells like it should be a helper function. Otherwise you're tracking this 0xFF everywhere, etc.